client/reducers/action.js

```diff
@@ -16,17 +16,21 @@ export const runCommands = () => {
   return (dispatch, getState) => {
     var interval = setInterval(() => {
       var state = getState()
-      if (state.running === false || state.robot.isAlive == false) {
+      if (!state.running){
+        clearInterval(interval)
+      } 
+      else if (state.robot.isAlive == false) {
+        dispatch(nextCommand("HAS_FINISHED"))
         clearInterval(interval)
-        return
       }
-      if (state.executeCommandIndex === state.commandQueue.length) {
+      else if (state.executeCommandIndex === state.commandQueue.length) {
         if (state.board[state.robot.positionY][state.robot.positionX] === 1) {
           dispatch(nextCommand("LEVEL_WON"))
```

Comment:
nextCommand is for dispatching actions from the command queue.

I know you didn't write this code but could you please create the action with a new function and use the LEVEL_WON variable.